Services/Devices.cs

```diff
@@ -74,54 +61,73 @@ public class Devices : IDevices
         // the simulation. When the simulation is running the registry is not used anymore.
         private const uint REGISTRY_LIMIT_REQUESTS = 1000;
 
-        // When working with batches, this is the max size that the batch insert and delete APIs allow
+        // When using bulk operations, this is the max number of devices that the registry APIs allow
         private const int REGISTRY_MAX_BATCH_SIZE = 100;
 
-        // When sending telemetry or other operations, wait only for 10 seconds. This setting sets how
-        // throttling affects the application. The default SDK value is 4 minutes, which causes high
-        // CPU usage.
-        private const int SDK_CLIENT_TIMEOUT = 10000;
-
         private readonly IIotHubConnectionStringManager connectionStringManager;
         private readonly ILogger log;
+        private readonly IServicesConfig config;
 
         private readonly bool twinReadsWritesEnabled;
         private string ioTHubHostName;
         private IRegistryManager registry;
         private int registryCount;
         private bool setupDone;
+        private string connString;
+        private string fixedDeviceKey;
 
         public Devices(
             IServicesConfig config,
             IIotHubConnectionStringManager connStringManager,
             IRegistryManager registryManager,
             ILogger logger)
         {
+            this.config = config;
             this.connectionStringManager = connStringManager;
+            this.connString = null;
             this.registry = registryManager;
             this.log = logger;
             this.twinReadsWritesEnabled = config.TwinReadWriteEnabled;
             this.registryCount = -1;
             this.setupDone = false;
         }
 
-        /// <summary>
-        /// Get IoTHub connection string from either the user provided value or the configuration
-        /// </summary>
+        // Get IoTHub connection string from either the user provided value or the configuration
         public void SetCurrentIotHub()
```

Comment:
How will this be called ? Can we consider calling this to setup iothub from constructor once ?

Reply:
It's called when the simulation runner starts. The setup cannot be executed in the ctor because it has external dependencies and could bring down the application in case of exceptions. It's also going to become an async method